src/main/java/site/coach_coach/coach_coach_server/category/service/CategoryService.java

```diff
@@ -0,0 +1,52 @@
+package site.coach_coach.coach_coach_server.category.service;
+
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import site.coach_coach.coach_coach_server.category.domain.Category;
+import site.coach_coach.coach_coach_server.category.dto.CreateCategoryRequest;
+import site.coach_coach.coach_coach_server.category.repository.CategoryRepository;
+import site.coach_coach.coach_coach_server.common.constants.ErrorMessage;
+import site.coach_coach.coach_coach_server.common.exception.AccessDeniedException;
+import site.coach_coach.coach_coach_server.routine.domain.Routine;
+import site.coach_coach.coach_coach_server.routine.exception.NoExistRoutineException;
+import site.coach_coach.coach_coach_server.routine.repository.RoutineRepository;
+import site.coach_coach.coach_coach_server.routine.service.RoutineService;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryService {
+	private final CategoryRepository categoryRepository;
+	private final RoutineService routineService;
+	private final RoutineRepository routineRepository;
+
+	public Long createCategory(CreateCategoryRequest createCategoryRequest, Long routineId, Long userIdByJwt) {
+		validateAccessCategory(routineId, userIdByJwt);
+
+		Routine routine = Routine.builder()
+			.routineId(routineId)
+			.build();
+
+		Category category = Category.builder()
+			.routine(routine)
+			.categoryName(createCategoryRequest.categoryName())
+			.isCompleted(false)
+			.build();
+
+		return categoryRepository.save(category).getCategoryId();
+	}
```

Comment:
```suggestion
	public Long createCategory(CreateCategoryRequest createCategoryRequest, Long routineId, Long userIdByJwt) {
    // 접근 권한 검증
    checkAccessToRoutine(routineId, userIdByJwt);

    // 루틴 객체를 직접 가져오기
    Routine routine = routineRepository.findById(routineId)
        .orElseThrow(() -> new NoExistRoutineException(ErrorMessage.NOT_FOUND_ROUTINE));

    // 카테고리 생성 및 저장
    Category category = Category.builder()
        .routine(routine)
        .categoryName(createCategoryRequest.categoryName())
        .isCompleted(false)
        .build();

    return categoryRepository.save(category).getCategoryId();
}
```

이렇게 하면 생성할 필요가 없지 않나.. 싶어서요!!

Reply:
저.. 제가 수정한 validate 메서드 명은,,
1n년의.. 의무 교육으로 인해 `validateAccessToCategory`가 더 자연스러운 표현같아서 습관적으로 저렇게 막 바꿔버렸습니다..